### PR TITLE
Perl Best Practices

### PR DESCRIPTION
**Pull request by Tux**
I've split them into small pieces

### CODE REVIEW

**Comment by linuxhw** [comment]
Looks like I can only merge this pull manually due to large number of changes, conflicts and non-acceptance of some intermediate commits affecting many lines of code (cherry-pick is not effective here). Sorry for this. I'll note your contribution in the commit message.

Thank you.

File: hw-probe.pl [hunk 1]
```diff
@@ -1138,7 +1138,7 @@ sub getRequest {
 sub saveProbe {
     my $To = $_[0];
 
-    $To=~s&/+\Z&&;
+    $To=~s{/+\Z}{};
```
Comment (linuxhw) [inline hw-probe.pl]: @Tux Is this compatible with Perl 5.8?

**Comment by Tux** [comment]
On Tue, 30 Oct 2018 18:25:17 +0000 (UTC), Linux Hardware Project
<notifications@github.com> wrote:

> linuxhw commented on this pull request.
> 
> > @@ -1138,7 +1138,7 @@ sub getRequest {  
>  sub saveProbe {
>      my $To = $_[0];
>  
> -    $To=~s&/+\Z&&;
> +    $To=~s{/+\Z}{};
> 
> @Tux Is this compatible with Perl 5.8?

Yes, it is:

$ perl5.8.8 -wle'$_="xyz";s{y}{ab};print'
xabz


-- 
H.Merijn Brand  http://tux.nl   Perl Monger  http://amsterdam.pm.org/
using perl5.00307 .. 5.29   porting perl5 on HP-UX, AIX, and openSUSE
http://mirrors.develooper.com/hpux/        http://www.test-smoke.org/
http://qa.perl.org   http://www.goldmark.org/jeff/stupid-disclaimers/

**Comment by linuxhw** [comment]
Merged manually by `git cherry-pick`.

**Pull request closed without merging**